src/conn.c

```diff
@@ -1465,3 +1448,32 @@ static int _conn_connect(xmpp_conn_t *conn,
 
     return 0;
 }
+
+static int _send_raw(xmpp_conn_t *conn, char *data, size_t len)
+{
+    xmpp_send_queue_t *item;
+
+    /* create send queue item for queue */
+    item = xmpp_alloc(conn->ctx, sizeof(xmpp_send_queue_t));
+    if (!item)
+        return XMPP_EMEM;
```

Comment:
another good catch, I'll add the free and debug statement!

Reply:
Also the parent functions never free'd it :)

`_send_raw()` always took ownership of the data (which is indicated by the `char*` parameter, if it would not take ownership, i.e. copy the data, it would be `const char*`) .

I'll also add a comment.